// Proto file representing the EDB Point Term class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "point_data.proto";
import "refs.proto";

service PointTerminalService {
  // Create a point term
  rpc Create (PointTermCreationMessage) returns (EDBObjMessage) {}

  rpc GetParameters(EDBObjMessage) returns (PointTermParamsMessage) {}

  rpc SetParameters(PointTermSetParamsMessage) returns (google.protobuf.BoolValue) {}
}

message PointTermCreationMessage {
  EDBObjMessage layout = 1;
  NetRefMessage net = 2;
  string name = 3;
  PointTermParamsMessage params = 4;
}

message PointTermParamsMessage {
  PointMessage point = 1;
  LayerRefMessage layer = 2;
}

message PointTermSetParamsMessage {
  EDBObjMessage term = 1;
  PointTermParamsMessage params = 2;
}